### ISSUE TITLE
**REDACTED**

### ISSUE THREAD

**Issue by XiXiongMaoXiong** [t-pankajkumar/DropBoxJava#2]

**Comment by t-pankajkumar** [comment]
What kind of help do you need?

**Comment by ThisIsM4R** [comment]
First of all thanks for giving time

I was reading the thread and noticed that someone said about OpenLoad which I use it too so you replied with a piece of code which was helpful but I couldn't figure out how to use that code, I mean.. I want to directly download the file to my OpenLoad and as I see your code does it with an HTTP request but I was thinking of using an FTP library to do that, anyway  my main problem is how can I run that code?
I have a Heroku account and as you said in the thread to create a new simple node.js app using the express framework but I couldn't find it.
Beside all of the long long talking of me, just help me download my torrents to my FTP server. 
And if you think OpenLoad isn't a good choice then I have other FTP servers too.. Just help me brother to have such thing.

**Comment by ThisIsM4R** [comment]
And by the way the developer of cloud torrent removed some functions of it in version 2 which sucks and he still keeps pointing people to thread #24 and I find no shit at #24 just the same old comments
And one thing else to be mentioned is that some guy added the SSH over FTP feature to the old version of cloud torrent which was written in Node.js but when I deployed it to heroku and modified the vars to my OpenLoad server with my username and password and port 21.. the app crashed and the logs keep saying that error in SSH, wrong username even tho I haven't entered any wrong info but I'm not sure that OpenLoad supports secure FTP protocol or not. And it does the same error for Mega too

**Comment by t-pankajkumar** [comment]
I have never used ftp library before.
And the code which have provided doesn't upload file via HTTP it just sends the downloaded file URL to Openload there it will fetch and saves it in your account we don't have to send bit by bit.
I think it is the most efficient way to upload.
Are you aware of Openload Remote Upload feature?

**Comment by ThisIsM4R** [comment]
Yup I'm aware of that but you may know that Heroku doesn't provides much storage to download torrents in it and they don't let you to use their bandwidth since CloudTorrent uses the max bandwidth so they suspended five of my account for deploying cloudtorrent and on the other side files won't last at their server I mean if you start a 500 MB file download it will get deleted in the matter of seconds and the app will shutdown and disconnect
So I was saying that maybe I can directly remote upload it to openload instead of downloading it firstly to heroku then to openload which isn't a great idea and won't work that much

**Comment by t-pankajkumar** [comment]
Matter of seconds ?
No way i download around 4gb file it stays more than an hour for me, then upload it after completion to dropbox or download it to immediately.
If you want this is my site : 
https://pank.herokuapp.com/torAdd?magnet=<magnet link>
and watch it here https://pank.herokuapp.com
Please dont use it for bad things :)

**Comment by ThisIsM4R** [comment]
No man I'm not the porn looking dude in Torrents, I use torrents mostly for programming courses and games not shits like porn
But believe me when I tell you they delete the file immediately, yeah sometimes it stays for an hour or more than an hour for me too but not always I download 4GB stuff, the min size of what I download is around 10 and max is 110 GB 
Thank you for your site mate but I think you can't help or either didn't understand what I meant brother

**Comment by ThisIsM4R** [comment]
For example I started something in your server check it out

**Comment by ThisIsM4R** [comment]
See!! your app crashed and gives error

**Comment by t-pankajkumar** [comment]
yup

**Comment by ThisIsM4R** [comment]
NOW WHAT?

**Comment by t-pankajkumar** [comment]
restarted it .
whats the size of file

**Comment by ThisIsM4R** [comment]
around 14 GB if this downloaded with success then I'll delete it and will do another 100GB test
If you agree of course 
If not then never mind

**Comment by ThisIsM4R** [comment]
Yup.. same shit
Again it crashed man

**Comment by t-pankajkumar** [comment]
ok if heroku keeps your file.
But Webtorrent.js didn't provide api to get stream to upload ftp (i think).
Right now im working on java based torrent lib which provides stream.

**Comment by ThisIsM4R** [comment]
Have you heard about NanoBox?

**Comment by ThisIsM4R** [comment]
It's a cloud app platform just like Heroku I've signed up there last night for FREE
You might wanna try it
https://dashboard.nanobox.io

**Comment by ThisIsM4R** [comment]
But what about the old version of cloud torrent? It has a ssh ftp backend 
https://github.com/jpillora/node-torrent-cloud

**Comment by t-pankajkumar** [comment]
heroku might have flagged those repositories as illegal , so we have to choose another provider

**Comment by ThisIsM4R** [comment]
I don't know much about Nanobox but you have to try it 
Enter you email and after an hour of two they'll send you an invitation code and you're good to go

**Comment by ThisIsM4R** [comment]
It supports
PYTHON
PHP
NODEJS
RUBY
JAVASCRIPT
GO
ELIXIR
JAVA
.NET
SCALA
RUST
ERLANG
PERL
CLOJURE
C

**Comment by t-pankajkumar** [comment]
I'll take a look at that,
May be this weekend will work on that.

**Comment by ThisIsM4R** [comment]
Anyway I'm waiting for the good news of your java-based torrent lib
But I hope you don't make it like J Pillora's shit and being greedy to answer people
Anyway just drop a comment here if you wanted to make me aware of your project

A million thanks again for giving you time to talk 

Stay awesome! :)

**Comment by t-pankajkumar** [comment]
Sure :)

**Comment by t-pankajkumar** [comment]
Yes i haven't tried that repo yet, i have to take a look at it.

**Comment by ThisIsM4R** [comment]
And can you take a look back at this old thing:
https://github.com/jpillora/node-torrent-cloud

And see why the SSH isn't working for OpenLoad?

**Comment by ThisIsM4R** [comment]
Hey mate, is your weekend arrived or what? Cause it has been 19 DAYS since I'm waiting for you 
Besides can I have your Email address in order to talk about something I've found

**Comment by t-pankajkumar** [comment]
Busy with my other works ....

**Comment by ThisIsM4R** [comment]
Oh OK, mind if you drop your email here?
